### PR TITLE
Fetch kt collections

### PR DESCRIPTION
**Pull request by SHeimlich**
PR number 2 to make Arcs so fetch!

- Made Collections no longer extend Iterable
- Added fetchAll() method that returns all entities in a set
- Added iterator() method to Collections

### CODE REVIEW

File: java/arcs/sdk/Handle.kt [hunk 1]
```diff
@@ -37,12 +37,18 @@ interface WritableSingleton<T : Entity> : Handle {
...
-interface ReadableCollection<T : Entity> : Handle, Iterable<T> {
+interface ReadableCollection<T : Entity> : Handle {
     /** The number of elements in the collection. */
     val size: Int
 
     /** Returns true if the collection is empty. */
     fun isEmpty(): Boolean
+
+    /** Returns an Iterator */
+    fun iterator(): Iterator<T>
```
Comment (piotrswigon) [inline java/arcs/sdk/Handle.kt]: This should not be here. Replace usages of iterator() to fetchAll().iterator()

File: java/arcs/sdk/jvm/CollectionImpl.kt [hunk 2]
```diff
@@ -28,6 +28,10 @@ class CollectionImpl<T : Entity>(
 
     override fun iterator(): Iterator<T> = entities.iterator()
```
Comment (piotrswigon) [inline java/arcs/sdk/jvm/CollectionImpl.kt]: delete this please

File: particles/Tutorial/Kotlin/README.md [hunk 3]
```diff
@@ -406,7 +406,7 @@ class CollectionsParticle : AbstractCollectionsParticle() {
         // We begin by generating the list of models that should fill the template. Our template
         // has name ang age so we will use these names.
         val peopleList = mutableListOf<Map<String, Comparable<*>?>>()
-        handles.people.forEach { people ->
+        handles.inputData.fetchAll().forEach { people ->
```
Comment (piotrswigon) [inline particles/Tutorial/Kotlin/README.md]: why did the handle name change?

File: javatests/arcs/sdk/wasm/EntitySlicingTest.kt [hunk 4]
```diff
@@ -23,13 +23,13 @@ class EntitySlicingTest : AbstractEntitySlicingTest() {
         handles.s2.fetch()?.let { handles.res.store(Res("s2:${it.num.toInt()},${it.txt}")) }
         handles.s3.fetch()?.let { handles.res.store(Res("s3:${it.num.toInt()},${it.txt},${it.flg}")) }
 
-        for (e in handles.c1) {
+        for (e in handles.c1.iterator()) {
```
Comment (piotrswigon) [inline javatests/arcs/sdk/wasm/EntitySlicingTest.kt]: for (e in handles.c1.fetchAll()) here and below

File: java/arcs/sdk/Handle.kt (hunk 1 above)
Comment (piotrswigon) [reply java/arcs/sdk/Handle.kt]: iterator() is a method that should be on iterables, but we're not an iterable anymore.

Comment (SHeimlich) [reply java/arcs/sdk/Handle.kt]: Updated

File: particles/Tutorial/Kotlin/README.md (hunk 3 above)
Comment (SHeimlich) [reply particles/Tutorial/Kotlin/README.md]: This is what it is in the actual kt file, I just copy-pasted it over.

File: java/arcs/sdk/Handle.kt [hunk 5]
```diff
@@ -37,12 +37,15 @@ interface WritableSingleton<T : Entity> : Handle {
...
 /** A collection handle with read access. */
-interface ReadableCollection<T : Entity> : Handle, Iterable<T> {
+interface ReadableCollection<T : Entity> : Handle {
     /** The number of elements in the collection. */
     val size: Int
 
     /** Returns true if the collection is empty. */
     fun isEmpty(): Boolean
+
+    /** Returns a set with all the entities in the collection*/
```
Comment (jasonwyatt) [inline java/arcs/sdk/Handle.kt]:
```suggestion
    /** Returns a set with all the entities in the collection. */
```

File: java/arcs/sdk/jvm/CollectionImpl.kt [hunk 6]
```diff
@@ -26,7 +26,9 @@ class CollectionImpl<T : Entity>(
 
     override fun isEmpty(): Boolean = entities.isEmpty()
 
-    override fun iterator(): Iterator<T> = entities.iterator()
+    override fun fetchAll(): Set<T> {
+        return entities.toSet()
+    }
```
Comment (jasonwyatt) [inline java/arcs/sdk/jvm/CollectionImpl.kt]: Function expression?

File: java/arcs/sdk/wasm/WasmCollectionImpl.kt [hunk 7]
```diff
@@ -25,7 +25,9 @@ class WasmCollectionImpl<T : WasmEntity>(
     override val size: Int
         get() = entities.size
 
-    override fun iterator() = entities.values.iterator()
+    override fun fetchAll(): Set<T> {
+        return entities.values.toSet()
+    }
```
Comment (jasonwyatt) [inline java/arcs/sdk/wasm/WasmCollectionImpl.kt]: Function expression?

File: java/arcs/sdk/Handle.kt (hunk 5 above)
Comment (csilvestrini) [reply java/arcs/sdk/Handle.kt]: Not done?

Comment (SHeimlich) [reply java/arcs/sdk/Handle.kt]: Attempt #2 appears to have succeeded.

**Pull request merged**